src/Core/Resolvers/SqlQueryEngine.cs

```diff
@@ -233,8 +257,83 @@ private OkObjectResult FormatFindResult(JsonDocument jsonDoc, FindRequestContext
                                   path,
                                   nvc: context!.ParsedQueryString,
                                   after);
-            rootEnumerated = rootEnumerated.Append(nextLink);
-            return OkResponse(JsonSerializer.SerializeToElement(rootEnumerated));
+
+            // When there are extra fields present, they are removed before returning the response.
+            if (extraFieldsInResponse.Count > 0)
+            {
+                rootEnumerated = RemoveExtraFieldsInResponseWithMultipleItems(rootEnumerated, extraFieldsInResponse);
+            }
+
+            return OkResponse(JsonSerializer.SerializeToElement(rootEnumerated.Append(nextLink)));
```

Comment:
Now that `rootEnumerated` is a list, you can use `Add()`. Using `Append()` creates a new list: https://stackoverflow.com/a/52355756/18174950